api/src/OpenTelemetry/Logging/Core.hs

```diff
@@ -101,16 +112,12 @@ makeLogger
 makeLogger loggerProvider loggerInstrumentationScope = Logger {..}
 
 
-{- | Emits a LogRecord with properties specified by the passed in Logger and LogRecordArguments.
-If observedTimestamp is not set in LogRecordArguments, it will default to the current timestamp.
-If context is not specified in LogRecordArguments it will default to the current context.
--}
-emitLogRecord
+createImmutableLogRecord
   :: (MonadIO m)
   => Logger
   -> LogRecordArguments body
-  -> m (LogRecord body)
-emitLogRecord Logger {..} LogRecordArguments {..} = do
+  -> m (ImmutableLogRecord body)
+createImmutableLogRecord logger@Logger {..} LogRecordArguments {..} = do
```

Comment:
`createImmutableLogRecord` is essentially the same function that `emitLogRecord` was before. This is a helper function to separate the `IORef` logic from the `LogRecord` creation logic.